initrd/bin/kexec-select-boot

```diff
@@ -1,8 +1,82 @@
 #!/bin/sh
 # Generic configurable boot script via kexec
-set -e -o pipefail
-. /tmp/config
-. /etc/functions
+
+usage() {
+	cat <<End-of-message
+Usage
+=====
+
+Generic configurable boot script via kexec
+	
+$0 -b /boot [options]
+	
+Options
+-------
+
+### -a
+
+Anything provided to the -a argument is appended to the kernel command line (usually comes from parsing the grub, syslinux, BLS files). Ex.
+
+	-a 'export CONFIG_BOOT_KERNEL_ADD="intel_iommu=on intel_iommu=igfx_off"'
+
+### -b
+
+path to boot files.  e.g.  -b /boot
+
+### -c = '*.cfg'
+
+specify the grub config file(s)
+
+### -d
+
+set the params device.  defaults to same as -b
+
+### -f
+
+force boot by considering hash and rollback checks as good
+
+### -g
+
+use graphical menus.  default is text only.
+
+### -h
+
+show help
+
+### -i
+
+Use this parameter to skip the verification of boot hashes and rollback counter.  This will boot a potentially compromised system.
+
+### -m
+
+show a boot menu
+
+### -o <initrd>
+
+use the specified initrd for booting
+
+### -p
+
+set the params folder.  defaults to same as -b
+
+### -r <configurations to remove>
+
+Anything provided to the -r argument is removed from the kernel command line. Ex.
+
+	-r 'export CONFIG_BOOT_KERNEL_REMOVE="quiet"'
+
+The 'quiet' option will hide needed information from the user.  In this case 'quiet' will be removd from the kernel configuration before it is booted.
```

Comment:
Same clarifications needs to be specified here.
Leaving -r 'export CONFIG_BOOT_KERNEL_REMOVE='quiet"' is misleading. User needs to understand where such parameter is coming from, and how to pass it directly to command line in case of testing and debugging. Otherwise, this documentatio nis aimed at users trying to understand how it works, and reading this line with export is confusing to me, so not sure it would be helpful, but more confusing for other users.

As a general guideline, in code documentation should state where command arguments are coming from with a really minimalist perspective. Otherwise we open a can of worm, as in here, where trying to cover more leads to more maintenance.

We have to remember that calling those directly is rarely ever needed, and called generally from gui-init, where the output of the transformation is actually given on screen as a result of the manipulations.

So if that part needs more documentation, in my opinion, there might be a bigger need to augment output on console where echo are talking from themselves for people who will actually debug the code and output of it, where general, high level documentation should state the higher level functionning of Heads without going in detail, where code should speak more about itself.

Reply:
I didn't touch kexec-select-boot bc we were still working on kexec-boot.